src/apps/feature_based_matching/feature_based_localizer.cpp

```diff
@@ -0,0 +1,81 @@
+// Created by O.Vysotska in 2023
+
+#include "online_localizer/online_localizer.h"
+#include "database/idatabase.h"
+#include "database/list_dir.h"
+#include "database/online_database.h"
+#include "features/cnn_feature.h"
+#include "features/ifeature.h"
+#include "online_localizer/path_element.h"
+#include "relocalizers/lsh_cv_hashing.h"
+#include "tools/config_parser/config_parser.h"
+
+#include <glog/logging.h>
+
+#include <iostream>
+#include <memory>
+#include <string>
+
+namespace loc = localization;
+
+std::vector<std::unique_ptr<loc::features::iFeature>>
+loadFeatures(const std::string &path2folder) {
+  LOG(INFO) << "Loading the features to hash with LSH.";
+  std::vector<std::string> featureNames =
+      loc::database::listProtoDir(path2folder, ".Feature");
+  std::vector<std::unique_ptr<loc::features::iFeature>> features;
+
+  for (size_t i = 0; i < featureNames.size(); ++i) {
+    features.emplace_back(
+        std::make_unique<loc::features::CnnFeature>(featureNames[i]));
+    fprintf(stderr, ".");
+  }
+  fprintf(stderr, "\n");
+  LOG(INFO) << "Features were loaded and binarized";
+  return features;
+}
+
+int main(int argc, char *argv[]) {
+  google::InitGoogleLogging(argv[0]);
+  FLAGS_logtostderr = 1;
+  LOG(INFO) << "===== Online place recognition with LSH ====\n";
+
+  if (argc < 2) {
+    printf("[ERROR] Not enough input parameters.\n");
+    printf("Proper usage: ./cost_matrix_based_matching_lsh config_file.yaml\n");
+    exit(0);
```

Comment:
nit: Follow up on above, if a program terminates with an error it probably should not return code `0` as it will be interpreted as "program terminated correctly".